core/trino-main/src/test/java/io/trino/sql/query/QueryAssertions.java

```diff
@@ -866,13 +872,30 @@ VALUES ROW(%s)
                 fail("Mismatched results between interpreter and evaluation engine: %s vs %s".formatted(full.value(), withConstantFolding.value()));
             }
 
-            return new Result(full.type(), full.value);
+            return new Result(full.type(), full.value, full.expression);
+        }
+
+        private static Optional<Expression> extractExpressionWithBindings(Plan plan)
```

Comment:
What's the purpose of this method and the one below?

Reply:
To get final expression how value was produced.